### PR TITLE
Issue 1141: Definition M

### PR DESCRIPTION
**Pull request by lucasmbrown-usds**
I believe this is ready for merge. Both `generate-score-post` and `geo-score` run fine. 

Here is the git diff between the file for Definition L and the file for Definition M:
[diff.txt](https://github.com/usds/justice40-tool/files/7884086/diff.txt)

### CODE REVIEW

File: data/data-pipeline/data_pipeline/score/score_l.py [hunk 1]
```diff
@@ -120,54 +120,6 @@ def _increment_total_eligibility_exceeded(
             axis=1, skipna=True
         )
 
-    def add_columns(self) -> pd.DataFrame:
```
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/score/score_l.py:123]: Just moving this to the end of the class to follow the convention of putting all private class methods before public class methods. I didn't change any part of the code here.

File: data/data-pipeline/data_pipeline/ipython/scoring_comparison.ipynb [hunk 2]
```diff
@@ -61,7 +62,10 @@
     "# Set some global parameters\n",
     "DATA_DIR = pathlib.Path.cwd().parent / \"data\"\n",
     "TEMP_DATA_DIR = DATA_DIR / \"tmp\"\n",
-    "COMPARISON_OUTPUTS_DIR = DATA_DIR / \"comparison_outputs\"\n",
+    "\n",
+    "time_str = time.strftime(\"%Y%m%d-%H%M%S\")\n",
+    "\n",
+    "COMPARISON_OUTPUTS_DIR = DATA_DIR / \"comparison_outputs\" / time_str\n",
```
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/ipython/scoring_comparison.ipynb:68]: Creating a unique output directory per run.

File: data/data-pipeline/data_pipeline/ipython/scoring_comparison.ipynb [hunk 3]
```diff
@@ -1001,24 +1053,28 @@
...
+    "        fields_to_group_by,\n",
     "        dropna=False,\n",
     "    )\n",
     "\n",
-    "    # Run the comparison function on the groups.\n",
-    "    comparison_df = grouped_df.mean().reset_index()\n",
+    "    # Take the mean of all fields.\n",
+    "    comparison_df = grouped_df.mean()\n",
+    "\n",
+    "    # Also add in the count of census tracts.\n",
```
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/ipython/scoring_comparison.ipynb:1071]: All these updates in this section are to add in the count of census tracts per groupby.

File: data/data-pipeline/data_pipeline/ipython/scoring_comparison.ipynb (hunk 2 above)
Comment (esfoobar-usds) [reply data/data-pipeline/data_pipeline/ipython/scoring_comparison.ipynb:68]: Nice!

**Review by esfoobar-usds — APPROVED** [review]
🍏 

@lucasmbrown-usds please go ahead and merge to `main` so @vim-usds and I can test locally

**Pull request merged**